vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java

```diff
@@ -611,6 +595,49 @@ else if (!decksToDelete.isEmpty()) {
   }
 
 
+  /**
+   * Find the Vassal DrawPile that matches a given Deck.
+   * Deck Names may have been repeated (ill-advised). DrawPiles may have been added to or removed from the module
+   * or may exist on Boards that are not active in the current game
+   * If a DrawPile exists with a matching Deck Name on the same Map, return it.
+   * Otherwise, return the first DrawPile found with a matching name
+   *
+   * @param deck Deck to search for
+   * @return     Matching DrawPile
+   */
+  private DrawPile findDrawPile(Deck deck) {
+    DrawPile first = null;
+    DrawPile matching = null;
+    final String deckName = deck.getDeckName();
+    for (final DrawPile pile : getModuleDrawPiles()) {
+      if (deckName.equals(pile.getAttributeValueString(SetupStack.NAME))) {
+
+        // If drawPile is owned by a specific board, then we can only match it if that board is active in this game
+        final String pileBoardName = pile.getOwningBoardName();
+        if (pileBoardName != null) {
+          if (deck.getMap().getBoardByName(pileBoardName) == null) {
+            continue;
+          }
+
+          // If the drawPile is on a map that doesn't have its current owning board active, then we
+          // cannot match that drawPile.
+          if (pile.getMap().getBoardByName(pile.getOwningBoardName()) == null) {
+            continue;
+          }
+        }
+
+        if (first == null) {
+          first = pile;
+          if (deck.getMap().equals(pile.getMap())) {
```

Comment:
Am I misunderstanding something here, or should this `if` statement be outside of the `(first == null)` check, instead of within it?  Is it not possible that you would find a match on the same map *after* finding a match on an earlier map?  As written, it looks to me like it stops checking for a matching map once it's found any matching name.

Reply:
Thanks Jonathon, my cut and paste hatchet job got out of hand. That code makes no sense at all. I have committed a new version.